apps/web/src/modules/proposal/components/ProposalDescription/MilestoneDetails.tsx

```diff
@@ -81,58 +108,117 @@ export const MilestoneDetails = ({
+    .map((x: string) => formatEther(BigInt(x)))
+
   const handleReleaseMilestone = async (index: number) => {
-    // TODO: Implement release milestone
-    console.log(`Milestone ${index} release attempted`)
+    const isClientGoverner = isAddressEqual(
+      clientAddress as AddressType,
+      addresses.governor as AddressType
+    )
+
+    if (!isClientGoverner) {
+      router.replace(SAFE_APP_URL)
+      return
+    }
+
+    removeAllTransactions()
+
+    const releaseMilestone = {
+      target: INVOICE_ADDRESS as AddressType,
+      functionSignature: 'release(_milestone)',
+      calldata: encodeFunctionData({
+        abi: RELEASE_FUNCTION_ABI,
+        functionName: 'release',
+        args: [index],
+      }),
+      value: '',
+    }
+
+    const releaseEscrowTxnData = {
+      type: TransactionType.RELEASE_ESCROW_MILESTONE,
+      summary: `Release Milestone #${index + 1} for ${invoiceData?.title}`,
+      transactions: [releaseMilestone],
+    }
+
+    addTransaction(releaseEscrowTxnData)
+
+    // Navigate to review page with release txn loaded
+    router.push({
+      pathname: `/dao/[network]/[token]/proposal/review`,
+      query: {
+        network: router.query?.network,
+        token: router.query?.token,
+      },
+    })
   }
 
-  const milestonesDetails = milestones?.map((milestone, index) => {
-    return {
-      title: (
-        <Text>
-          {index + 1}. {milestone.title}
-        </Text>
-      ),
-      description: (
-        <Stack gap={'x5'}>
-          <Stack direction={'row'} align={'center'} justify={'space-between'}>
-            <Text variant="label-xs" color="tertiary">
-              {'Amount: ' + milestoneAmount[index] + ' ETH'}
-            </Text>
-            <Text variant="label-xs" color="tertiary">
-              {'Due by: ' + new Date(milestone?.endDate as number).toLocaleDateString()}
-            </Text>
-          </Stack>
-          <Text>{milestone.description || 'No Description'}</Text>
-          <Stack>
-            {milestone.documents?.map((doc, index) => {
-              if (doc.src) {
-                return (
-                  <OptionalLink
-                    enabled={true}
-                    href={doc?.src.replace('ipfs://', 'https://ipfs.io/ipfs/')}
-                  >
-                    {doc.type === 'ipfs'
-                      ? doc?.src.replace('ipfs://', 'https://ipfs.io/ipfs/')
-                      : doc.src}
-                  </OptionalLink>
-                )
-              }
-            })}
+  const renderMilestoneButton = (index: number, isReleased: boolean, isNext: boolean) => {
+    if (isReleased) {
+      return (
+        <Button variant="secondary" disabled>
+          <Icon id="checkInCircle" />
+          Milestone Released
+        </Button>
+      )
+    }
+
+    return (
+      <Button
+        variant={isNext ? 'primary' : 'secondary'}
+        disabled={!isNext}
+        onClick={() => isNext && handleReleaseMilestone(index)}
```

Comment:
you can show not disabled for any milestone after or including the current milestone right? Just need to let the user know that all milestones upto that point would be released.